lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -0,0 +1,479 @@
+                          [1, 0, 0, 0, 0],
+                          [1, 0, 0, 0, 0],
+                          [1, 1, 1, 1, 1]]])
+        self.weights_cube.data = np.ma.masked_array(self.weights_cube.data,
+                                                    mask=mask)
+        plugin = CollapseMaskedNeighbourhoodCoordinate(
+            "topographic_zone", weights=self.weights_cube)
+        result = plugin.process(self.nbhooded_cube)
+        expected_result = np.array([[0.0, 0.0, 0.0, 0.2, 0.2],
+                                    [0.0, 0.0, 0.2, 0.2, 0.2],
+                                    [0.0, 0.19, 0.2, 0.2, 0.2],
+                                    [0.0, 0.2, 0.2, 0.19, 0.2],
+                                    [0.0, 0.0, 0.0, 0.0, 0.0]])
+        expected_mask = np.array([[True, True, True, False, False],
+                                  [True, True, False, False, False],
+                                  [True, False, False, False, False],
+                                  [True, False, False, False, False],
+                                  [True, True, True, True, True]])
+
+        self.assertArrayAlmostEqual(expected_result, result.data.data)
+        self.assertArrayAlmostEqual(expected_mask, result.data.mask)
+        self.assertEqual(expected_mask.dtype, bool)
+        self.assertIsInstance(expected_result, np.ndarray)
+
+    def test_multidemensional_neighbourhood_input(self):
+        """Test that we can collapse the right dimension when there
+           are additional leading dimensions like threshold."""
+        nbhooded_cube = add_dimensions_to_cube(self.nbhooded_cube,
+                                               {"threshold": 3})
+        expected_result = np.array([[[0.12, 0.13, 0.2, 0.2, 0.2],
+                                     [0.13, 0.17, 0.2, 0.2, 0.2],
+                                     [0.17, 0.19, 0.2, 0.2, 0.2],
+                                     [0.2, 0.2, 0.2, 0.19, 0.15],
+                                     [0.2, 0.2, 0.2, 0.16, 0.12]],
+                                    [[0.12, 0.13, 0.2, 0.2, 0.2],
+                                     [0.13, 0.17, 0.2, 0.2, 0.2],
+                                     [0.17, 0.19, 0.2, 0.2, 0.2],
+                                     [0.2, 0.2, 0.2, 0.19, 0.15],
+                                     [0.2, 0.2, 0.2, 0.16, 0.12]],
+                                    [[0.12, 0.13, 0.2, 0.2, 0.2],
+                                     [0.13, 0.17, 0.2, 0.2, 0.2],
+                                     [0.17, 0.19, 0.2, 0.2, 0.2],
+                                     [0.2, 0.2, 0.2, 0.19, 0.15],
+                                     [0.2, 0.2, 0.2, 0.16, 0.12]]])
+        result = self.plugin.process(nbhooded_cube)
+        self.assertArrayAlmostEqual(expected_result, result.data)
+
+    def test_preserve_dimsensions_input(self):
+        """Test that the dimsensions on the output cube are the same as the
```

Comment:
"dimensions" (function name and docstring).  Also at line 457.

Reply:
Fixed. Apparently this is something I consistently mistype.